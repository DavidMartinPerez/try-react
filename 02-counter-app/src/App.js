import React from 'react';

const App = () => ( <h1>Probando app</h1> )

export default App;