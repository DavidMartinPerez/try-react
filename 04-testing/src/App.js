import React from 'react';
import PropTypes from 'prop-types';

const App = ( { saludo, nombre } ) => {


    return ( <h1> {saludo} { nombre } </h1> );
}

//Validaci√≥n de datos obligatorios
App.propTypes = {
    nombre: PropTypes.string.isRequired,
    saludo: PropTypes.string
};

App.defaultProps = {
    saludo: 'Hola, soy'
}

export default App;