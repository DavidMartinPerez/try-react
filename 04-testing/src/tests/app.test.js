import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import App from '../App'

describe('Testeamos un componente', () => {
    test('debe de mostrar <App/> correctamente', () => {
        const name = 'Bruce'
        const wrapper = shallow( <App nombre={name} /> );

        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe mostrar el saludo introducido', () => {
        const nameTest = 'Parker';
        const saludoTest = 'Spiderman es'

        const wrapper = shallow( <App saludo={saludoTest} nombre={nameTest} /> );

        const textTest = wrapper.find( 'h1' ).text();

        expect( textTest ).toBe( `${saludoTest} ${nameTest}` );

    })
})
