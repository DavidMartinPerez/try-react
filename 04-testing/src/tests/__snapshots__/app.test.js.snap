// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testeamos un componente debe de mostrar <App/> correctamente 1`] = `
<h1>
  Hola, soy
   
  Bruce
</h1>
`;
